enterprise/backend/test/metabase_enterprise/audit_app/api/database_test.clj

```diff
@@ -0,0 +1,20 @@
+(ns metabase-enterprise.audit-app.api.database-test
+  (:require
+   [clojure.test :refer :all]
+   [metabase-enterprise.audit-db :as audit-db]
+   [metabase-enterprise.audit-db-test :as audit-db-test]
+   [metabase.models.permissions :as perms]
+   [metabase.public-settings.premium-features-test :as premium-features-test]
+   [metabase.test :as mt]))
+
+(deftest audit-db-unmodifiable-test
+  (premium-features-test/with-premium-features #{:audit-app}
+    (audit-db-test/with-audit-db-restoration
+      (audit-db/ensure-audit-db-installed!)
+      (testing "Neither admin nor regular users can modify the audit database"
+        (doseq [[verb path] [[:post "database/%d/unpersist"]
+                             [:put "database/%d"]
+                             [:get "database/%d/syncable_schemas"]]
```

Comment:
I think you can add `[:delete "database/%d"]` to this list to get the failing repro